app/commands/github/task/sync_task.rb

```diff
@@ -0,0 +1,50 @@
+module Github
+  class Task
+    class SyncTask
+      include Mandate
+
+      initialize_with :issue
+
+      def call
+        if closed? || claimed?
+          destroy_task
+        else
+          create_or_update_task
+        end
+      end
+
+      private
+      def closed?
+        issue.status == :closed
+      end
+
+      def claimed?
+        issue.labels.any? { |label| label.name == Github::IssueLabel.for_type(:status, :claimed) }
```

Comment:
If I'm understanding this correctly this is going to load all the labels into memory then check each for the name.

I think maybe SQL would be better to avoid the extract weight of query and object allocation:
```ruby
issues.labels.where(name: Github::IssueLabel.for_type(:status, :claimed)).exists?
```

However, because we also use `issue.labels` again below, it might be better to have what you have here and memoize `issue.labels` into a method for `labels` in this command? So your code is fine, but the `find_label_of_type` method is expensive as it'll rebuild the `issue.labels` each time.